synthtool/languages/python.py

```diff
@@ -93,6 +94,36 @@ def _get_sample_readme_metadata(sample_dir: Path) -> dict:
     return sample_metadata
 
 
+def detect_versions(
+    path: str = "./src", default_version: Optional[str] = None
+) -> List[str]:
+    """
+    Detects the versions a library has, based on distinct folders
+    within path. This is based on the fact that our GAPIC libraries are
+    structured as follows:
+
+    src/v1
+    src/v1beta
+    src/v1alpha
+
+    With folder names mapping directly to versions.
+
+    Returns: a list of the subdirectories; for the example above:
+      ['v1', 'v1alpha', 'v1beta']
+      If specified, the default_version is guaranteed to be listed last.
+      Otherwise, the list is sorted alphabetically.
+    """
+    versions = []
+    if os.path.isdir(path):
+        for directory in os.listdir(path):
+            if os.path.isdir(os.path.join(path, directory)):
+                versions.append(directory)
```

Comment:
I believe you could glob for directories that look like they have versions:

```py
>>> import glob
>>> from pathlib import Path
>>> list(Path("google").glob("**/*v[1-9]*"))
[PosixPath('google/cloud/automl_v1'), PosixPath('google/cloud/automl_v1beta1')]
```

Reply:
Fixed in https://github.com/googleapis/synthtool/pull/1114/commits/fb23307ccc88cc41f1dde056aeea6fd1632748fe